gflownet/utils/batch.py

```diff
@@ -0,0 +1,179 @@
+from collections import defaultdict
```

Comment:
Should this really be in `gflownet/utils`? It is an essential part of the library. I would personally just put in in `gflownet/batch.py` (and would do the same for the oracle, for that matter).

Reply:
I personally treat `gflownet/utils` as an essential part of the library, just maybe more technical :) We have buffer, logger, metrics there, it seems to me that batch goes under the same umbrella

Reply:
I do not have much experience in creating python packages so I do not know the conventions, but either way seems ok. Let's leave it here for now?